### PR TITLE
Added payment/payment CRUD, Events, Saga Support and Connection to Simulation Service

### PR DESCRIPTION
**Pull request by KiLanger**

### CODE REVIEW

File: src/payment.gql [hunk 1]
```diff
@@ -88,7 +106,20 @@ type PaymentInformationConnection {
 }
 
 type Query {
```
Comment (nk-coding) [inline src/payment.gql:215]: paymentInformations and payments should also (only or additionally) be located on the user type

File: src/payment.gql [hunk 2]
```diff
@@ -88,7 +106,20 @@ type PaymentInformationConnection {
...
+    first: Int = 2147483647
+
+    """Ordering"""
+    orderBy: PaymentOrder
+
+    """Filtering"""
+    filter: PaymentFilter
+  ): PaymentConnection!
 
   """ Retrieves all payment informations matching the fitler"""
```
Comment (nk-coding) [inline src/payment.gql]: fitler

File: src/payment-provider-connection/connector.service.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+   * Sends a request to the specified endpoint with the provided data.
+   * @param endpoint The endpoint to send the request to.
+   * @param data The data to send with the request.
+   * @returns An Observable that emits the AxiosResponse object.
+   * @throws An error if the request fails.
+   */
+  async send(endpoint: string, data: any): Promise<AxiosResponse> {
+    try {
+      const response = await this.httpService
+        .post(`http://localhost:7000/${endpoint}`, data)
```
Comment (nk-coding) [inline src/payment-provider-connection/connector.service.ts]: this looks like debug code

File: src/payment-provider-connection/payment-processors/invoice.service.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import { EventService } from 'src/events/events.service';
+import { PaymentService } from 'src/payment/payment.service';
+import { PaymentStatus } from 'src/shared/enums/payment-status.enum';
+import { ConnectorService } from '../connector.service';
+import { Cron } from '@nestjs/schedule';
+import { xDaysBackFromNow } from 'src/shared/utils/functions.utils';
+import { PaymentMethod } from 'src/payment-method/payment-method.enum';
+
+@Injectable()
+export class InvoiceService {
```
Comment (nk-coding) [inline src/payment-provider-connection/payment-processors/invoice.service.ts:14]: missing docs

Comment (nk-coding) [reply src/payment-provider-connection/payment-processors/invoice.service.ts:14]: and also all other services

File: src/payment-provider-connection/payment-provider-connection.service.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  constructor(
+    private readonly paymentInformationService: PaymentInformationService,
+    private readonly creditCardService: CreditCardService,
+    private readonly prepaymentService: PrepaymentService,
+    private readonly invoiceService: InvoiceService,
+    // initialize logger with service context
+    private readonly logger: Logger,
+  ) {}
+
+  startPaymentProcess(paymentMethod: PaymentMethod, id: string): Promise<any> {
```
Comment (nk-coding) [inline src/payment-provider-connection/payment-provider-connection.service.ts:37]: docs

File: src/payment/payment.service.ts [hunk 6]
```diff
@@ -1,25 +1,235 @@
...
+
+    return existingPayment;
+  }
+
+  /**
+   * Builds a query object based on the provided filter.
+   * @param filter - The filter object containing the criteria for the query.
+   * @returns The query object.
+   */
+  buildQuery(filter: any): any {
```
Comment (nk-coding) [inline src/payment/payment.service.ts]: types are needed here

**Review by nk-coding — CHANGES_REQUESTED** [review]

File: src/payment.gql (hunk 2 above)
Comment (KiLanger) [reply src/payment.gql]: What? It says Filter? :D

Comment (nk-coding) [reply src/payment.gql]: """ Retrieves all payment informations matching the fitler"""

File: src/payment.gql (hunk 1 above)
Comment (KiLanger) [reply src/payment.gql:215]: payments is a gropius issue:
https://frontend.gropius.duckdns.org/components/e069c868-c07f-4b9c-b2d7-ad6bb2257caf/issues/0dd9ca69-0abc-4db1-84b8-2e8ca317db45

**Review by nk-coding — APPROVED** [review]

**Pull request merged**